### DISCUSSION TITLE
Can track Korean in-app browser in more detail?

### DISCUSSION THREAD

**Discussion by amblerkr** [dubinc/dub#160] [Ideas]
![image](https://user-images.githubusercontent.com/67187038/222952606-18720138-81db-48f4-8d93-928c1276e161.png)
![image](https://user-images.githubusercontent.com/67187038/222952600-b52f937a-8ae2-45f3-a176-78643b192331.png)

![image](https://user-images.githubusercontent.com/67187038/222952323-2f81a643-fefe-40f0-9ecf-6eebef818d7a.png)
The short address service called 'Rebrandly' is the only one that tracks in-app browsers in Korea. I wonder if you can show me in more detail.

- Chrome
- **Samsung Mobile: Korean Search Browser**
- **Naver-search-app: Korean Search Engine**
- Safari mobile
- Chrome mobile
- **Whale-mobile: Korean Search Browser**
- **Kakaotalk-app: Korean Messenger**
- Microsoft edge
- Safari
- Android-webview
- **Daum-app: Korean Search Engine**

**Comment by amblerkr** [comment]
The great advantage of dub is that it does geographic tracking better than any other service. 
Some shortened address services mark Korea as another country. I raised a problem, but they said they couldn't fix it.
I admire Vercel's technology. Perfect for displaying the connection location. It is good to show the national flag.

If you can track browser in more detail, this is the perfect Shorten URL Service.

**Comment by steven-tey** [comment]
Interesting idea! I believe we rely on https://github.com/faisalman/ua-parser-js for the `userAgent` parser in Next.js Middleware – do you know if that supports Korean browsers? You can try that out in the User Agent Playground here: http://faisalman.github.io/ua-parser-js/

**Reply by amblerkr** [reply]
I'm not a developer, so I don't know. Is this what you mean? 
![image](https://user-images.githubusercontent.com/67187038/235277228-f24ee878-0542-4dfa-a658-d9a5d87e4913.png)

https://faisalman.github.io/ua-parser-js-docs/v2/api/ua-parser-js/get-browser.html
It looks like this 
> Kakao[Story/Talk],  Samsung Browser, Line(NAVER JAJAN), Whale Browser(NAVER)


South Korea is dominated by KAKAO(=Daum) and NAVER. 
I tested it now. I have the latest models of Galaxy and iPhone. 
I created a shorten URL and tested it on my phone.
![image](https://user-images.githubusercontent.com/67187038/235278005-a683ad6a-304d-4885-a4bd-609eec9787d3.png)
![image](https://user-images.githubusercontent.com/67187038/235278019-2037aed3-ce5e-4458-90fd-a3ac26189782.png)

- Samsung Mobile: Korean Search Browser ─ Support. 'Samsung Browser' displayed
https://play.google.com/store/apps/details?id=com.sec.android.app.sbrowser&hl=ko&gl=US

- Whale-mobile: Korean Search Browser ─ Support. 'Whale' displayed
https://play.google.com/store/apps/details?id=com.naver.whale&hl=ko&gl=US
- NAVER-app: Korean Search Engine ─ 'Chrome WebView' displayed
https://play.google.com/store/apps/details?id=com.nhn.android.search&hl=ko&gl=US

- Kakaotalk-app: Korean Messenger ─ 'Chrome WebView' displayed
https://play.google.com/store/apps/details?id=com.kakao.talk&hl=ko&gl=US
- Daum-app: Korean Search Engine ─  'Chrome WebView' displayed. In the [Referrers] entry, it will appear as 'm.daum.net'.
https://play.google.com/store/apps/details?id=net.daum.android.daum&hl=ko&gl=US
![image](https://user-images.githubusercontent.com/67187038/235278217-bea6edab-aa99-4068-adc6-eaf45523b5ef.png)

---
etc...
- Safari-app ─ 'Mobile Safari' displayed
